src/Service/URLSanitisationService.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace SilverStripe\StaticPublishQueue\Service;
+
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Environment;
+
+/**
+ * Class URLSanitisationService
+ * Provides an abstraction for populating and generating a list of
+ * URLs to be cached by StaticPublishQueue.
+ *
+ * @package SilverStripe\StaticPublishQueue\Service
+ */
+class URLSanitisationService
+{
+    use Configurable;
+
+    /**
+     * Enable to force all URLs to be changed to https
+     *
+     * @var bool
+     * @config
+     */
+    private static $force_ssl = false;
```

Comment:
This is needed because CLI is not running with SSL. This is what we have in config:

```
if (!Director::is_cli()) {
    Director::forceSSL();
}
```

This is because if you force SSL to whole website, cli runners will bail out on HTTPS redirects.